play-json/jvm/src/main/scala/play/api/libs/json/EnvReads.scala

```diff
@@ -415,6 +419,78 @@ trait EnvReads {
     case _ => JsError(JsonValidationError("error.expected.jsstring"))
   }
 
+  private def javaDurationNumberReads(unit: TemporalUnit, esuffix: String) =
+    Reads[JDuration] {
+      case JsNumber(n) if !n.ulp.isWhole =>
+        JsError(s"error.invalid.$esuffix")
+
+      case JsNumber(n) => JsSuccess(JDuration.of(n.toLong, unit))
+      case _ => JsError(s"error.expected.$esuffix")
+    }
+
+  /**
+   * Deserializer of Java Duration from an integer (long) number,
+   * using the specified temporal unit.
+   */
+  def javaDurationNumberReads(unit: TemporalUnit): Reads[JDuration] =
+    javaDurationNumberReads(unit, "duration")
+
+  /** Deserializer of Java Duration from a number of milliseconds. */
+  val javaDurationMillisReads: Reads[JDuration] =
+    javaDurationNumberReads(ChronoUnit.MILLIS)
+
+  /**
+   * Deserializer of Java Duration, from either a time-based amount of time
+   * (string representation such as '34.5 seconds'),
+   * or from a number of milliseconds (see [[javaDurationMillisReads]]).
+   */
+  implicit val DefaultJavaDurationReads: Reads[JDuration] = Reads[JDuration] {
+    case JsString(repr) => try {
+      JsSuccess(JDuration.parse(repr))
+    } catch {
+      case _: Throwable => JsError("error.invalid.duration")
+    }
+
+    case js => javaDurationMillisReads.reads(js)
+  }
+
+  /**
+   * Derializer of Java Period from an integer (long) number,
+   * using the specified temporal unit.
+   */
+  def javaPeriodNumberReads(unit: TemporalUnit): Reads[Period] = {
+    val underlying = javaDurationNumberReads(unit, "period")
+
+    Reads[Period] { js =>
+      underlying.reads(js).flatMap { duration =>
+        try {
+          JsSuccess(Period.ofDays(duration.toDays.toInt))
+        } catch {
+          case e: Throwable => JsError("error.invalid.period")
```

Comment:
> An exception while parsing period should be represented with the JsResult not letting the exception bubbling up.

I disagree. `JsResult` is not a `Try`. `JsError` really only has the capacity to represent JSON parsing and validation errors, and I'm pretty sure that's what users expect of it. By presenting any exception as if it were invalid input, you're potentially masking other more dangerous errors.